haskell/repl.bzl

```
@@ -316,12 +315,25 @@ def _compiler_flags_and_inputs(hs, repl_info, static = False, path_prefix = ""):
         cc_library_files = _concat(get_library_files(hs, cc_libraries_info, cc_libraries))
     else:
         cc_library_files = get_ghci_library_files(hs, cc_libraries_info, cc_libraries)
-    link_libraries(cc_library_files, args)
+
+    # REPL scripts `cd` into the workspace root. Depending on their provenance,
+    # some C libraries' files may be in subdirectories relative to the execroot.
+    # External static C library dependencies are an example of this -- unchanged
+    # we may end up with paths like `external/some_dependency/lib` and/or
+    # `bazel-out/k8-fastbuild/bin/_solib_k8/...`; the former containing the
+    # archive (`.a`) file we want, but being relative to the execroot, and the
+    # latter being relative to the workspace root but only containing dynamic
+    # libraries.
+    #
+    # We fix this by identifying these libraries when generating linker flags and
+    # prefixing them so that they will be visible when the REPL runs.
+    link_libraries(cc_library_files, args, path_prefix = "$RULES_HASKELL_EXEC_ROOT/")
```

```suggestion
    link_libraries(cc_library_files, args, path_prefix = path_prefix + "/")
```
`path_prefix` is already an argument to `_compiler_flags_and_inputs`.